fitbenchmarking/parsing/nist_data_functions.py

```diff
@@ -26,8 +26,10 @@ def nist_func_definitions(function, startvals):
     function_defs = []
 
     # Create a function def for each starting set in startvals
+    if not is_safe(function_scipy_format):
+        raise ValueError('Error while sanitizing input')
+    exec("def fitting_function(x, " + param_names + "): return " + function_scipy_format)
```

Comment:
Would an empty dictionary not be even safer?

Reply:
No, because unless you explicitly remove `__builtins__` (like `__import__`) then you can still access these even after globals are removed:
```
In [9]: eval("__import__('os').getcwd()", {'__builtins__':{}})                                                                                                                                                                                
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-9-b5b98649c412> in <module>
----> 1 eval("__import__('os').getcwd()", {'__builtins__':{}})

<string> in <module>

NameError: name '__import__' is not defined

In [10]: eval("__import__('os').getcwd()")                                                                                                                                                                                                    
Out[10]: '/Users/thomasfarmer/Documents/GitHub/fitbenchmarking'
```
Setting globals to an empty dictionary just guarantees that if you have already imported a module it won't be accessible.  However as you can just import anyway, it's not much protection.

Reply:
Thanks for the explanation! I'll update